packages/website/src/components/editor/Editor.tsx

```diff
@@ -0,0 +1,329 @@
+import React from 'react';
+import type Monaco from 'monaco-editor';
+
+import { sandboxSingleton } from './loadSandbox';
+import { debounce } from '../lib/debounce';
+
+import type { Sandbox, SandboxConfig } from '../../vendor/sandbox';
+import type { WebLinter, TSESTree } from '@typescript-eslint/website-eslint';
+import type { ConfigModel, RuleDetails } from '../types';
+import { lintCode, LintCodeAction } from './lintCode';
+import { action } from './action';
+
+interface EditorProps extends ConfigModel {
+  readonly darkTheme: boolean;
+  readonly decoration?: TSESTree.Node | null;
+  readonly onChange?: (value: string) => void;
+  readonly onASTChange?: (
+    value: string | TSESTree.Program,
+    position: Monaco.Position | null,
+  ) => void;
+  readonly onLoadRule?: (value: RuleDetails[]) => void;
+  readonly onSelect?: (position: Monaco.Position) => void;
+  readonly onLoaded?: (tsVersions: readonly string[]) => void;
```

Comment:
The `Editor` is only called in one place, so all these optional ones don't actually need to be optional. Which should then mean that `if (` statements for them below shouldn't be necessary.